benchmarking/commons/hpo_main_local.py

```diff
@@ -151,58 +138,65 @@ def create_objects_for_tuner(
-        max_size_data_for_model=args.max_size_data_for_model,
+        max_size_data_for_model=configuration.max_size_data_for_model,
         benchmark=benchmark,
-        extra_args=None if extra_args is None else extra_metadata(args, extra_args),
+        extra_metadata=extra_job_metadata,
     )
     return dict(
         scheduler=scheduler,
         stop_criterion=stop_criterion,
         n_workers=benchmark.n_workers,
-        tuner_name=args.experiment_tag,
+        tuner_name=configuration.experiment_tag,
         metadata=metadata,
-        save_tuner=args.save_tuner,
+        save_tuner=configuration.save_tuner,
     )
 
 
-def main(
+def start_local_benchmark(
+    configuration: ConfigDict,
     methods: MethodDefinitions,
     benchmark_definitions: RealBenchmarkDefinitions,
-    extra_args: Optional[ExtraArgsType] = None,
-    map_extra_args: Optional[MapExtraArgsType] = None,
     post_processing: Optional[PostProcessingType] = None,
+    map_method_args: Optional[MapMethodArgsType] = None,
+    tuning_job_metadata: Optional[DictStrKey] = None,
 ):
     """
-    Runs sequence of experiments with local backend sequentially. The loop runs
-    over methods selected from ``methods`` and repetitions, both controlled by
-    command line arguments.
+    Runs sequence of experiments with local backend sequentially.
+    The loop runs over methods selected from ``methods`` and repetitions,
 
-    ``map_extra_args`` can be used to modify ``method_kwargs`` for constructing
+    ``map_method_args`` can be used to modify ``method_kwargs`` for constructing
     :class:`~benchmarking.commons.baselines.MethodArguments`, depending on
-    ``args`` returned by :func:`parse_args` and the method. Its signature is
-    :code:`method_kwargs = map_extra_args(args, method, method_kwargs)`, where
-    ``method`` is the name of the baseline.
+    ``configuration`` and the method. This allows for extra flexibility to specify specific arguments for chosen methods
+    Its signature is :code:`method_kwargs = map_method_args(configuration, method, method_kwargs)`,
+    where ``method`` is the name of the baseline.
 
-    :param methods: Dictionary with method constructors
+    :param configuration: ConfigDict with parameters of the benchmark.
+        Must contain all parameters from LOCAL_LOCAL_BENCHMARK_REQUIRED_PARAMETERS
+    :param methods: Dictionary with method constructors.
     :param benchmark_definitions: Definitions of benchmarks; one is selected from
         command line arguments
-    :param extra_args: Extra arguments for command line parser. Optional
-    :param map_extra_args: See above, optional
     :param post_processing: Called after tuning has finished, passing the tuner
         as argument. Can be used for postprocessing, such as output or storage
         of extra information
+    :param map_method_args: See above, optional
+    :param tuning_job_metadata: Metadata added to the tuner, can be used to manage results
```

Comment:
I want to make it clear that the metadata is for the tuner to make it easier to follow (for developers too), otherwise it can be unclear what is the job.

Reply:
But the important point here is that this metadata is on top of defaul metadata, and it is just the metadata related to this user-defined additional arguments.

Reply:
Yes